extensions/spring-scheduled/deployment/src/main/java/io/quarkus/spring/scheduling/deployment/SpringSchedulingProcessor.java

```diff
@@ -0,0 +1,175 @@
+        AnnotationStore annotationStore = beanRegistrationPhase.getContext().get(BuildExtension.Key.ANNOTATION_STORE);
+
+        for (BeanInfo bean : beanRegistrationPhase.getContext().beans().classBeans()) {
+            ClassInfo classInfo = bean.getTarget().get().asClass();
+            for (MethodInfo method : classInfo.methods()) {
+                List<AnnotationInstance> schedules = null;
+                AnnotationInstance scheduledAnnotation = annotationStore.getAnnotation(method, SPRING_SCHEDULED);
+                if (scheduledAnnotation != null) {
+                    schedules = Collections.singletonList(scheduledAnnotation);
+                } else {
+                    AnnotationInstance scheduledsAnnotation = annotationStore.getAnnotation(method, SPRING_SCHEDULES);
+                    if (scheduledsAnnotation != null) {
+                        schedules = new ArrayList<>();
+                        for (AnnotationInstance scheduledInstance : scheduledsAnnotation.value().asNestedArray()) {
+                            schedules.add(scheduledInstance);
+                        }
+                    }
+                }
+                processSpringScheduledAnnotation(scheduledBusinessMethods, bean, method, schedules);
+
+            }
+
+        }
+    }
+
+    void processSpringScheduledAnnotation(BuildProducer<ScheduledBusinessMethodItem> scheduledBusinessMethods,
+            BeanInfo bean, MethodInfo method, List<AnnotationInstance> scheduledAnnotations) {
+        List<AnnotationInstance> schedules = new ArrayList<>();
+        if (scheduledAnnotations != null) {
+            for (AnnotationInstance scheduledAnnotation : scheduledAnnotations) {
+                List<AnnotationValue> springAnnotationValues = scheduledAnnotation.values();
+                List<AnnotationValue> confValues = new ArrayList<>();
+                if (!springAnnotationValues.isEmpty()) {
+                    if (annotationsValuesContain(springAnnotationValues, "fixedRate")
+                            || annotationsValuesContain(springAnnotationValues, "fixedRateString")) {
+                        confValues.add(buildEveryParam(springAnnotationValues));
+                        if (annotationsValuesContain(springAnnotationValues, "initialDelay")
+                                || annotationsValuesContain(springAnnotationValues, "initialDelayString")) {
+                            confValues.addAll(buildDelayParams(springAnnotationValues));
+                        }
+
+                    } else if (annotationsValuesContain(springAnnotationValues, "fixedDelay")) {
+                        throw new IllegalStateException(
+                                "Unsupported @Scheduled param fixedDelay");
+                    } else if (annotationsValuesContain(springAnnotationValues, "cron")) {
+                        if (annotationsValuesContain(springAnnotationValues, "initialDelay")) {
+                            throw new IllegalStateException(
```

Comment:
We should have a consistent format for the error messages.  Also, we should probable use `IllegalArgumentException` instead of `IllegalStateException`

Reply:
I changed the exception type and homogenized the message, wdyt @geoand

Reply:
:+1: